src/components/organisms/EntityDetails/EntityDetailsContent/Settings/SettingsTests/SettingsTests.tsx

```diff
@@ -74,161 +68,170 @@ const SettingsTests: React.FC<{openDefinition(): void}> = ({openDefinition}) =>
+  const wasTouched = steps !== initialSteps;
 
   useEffect(() => {
-    if (currentSteps !== initialSteps) {
-      setCurrentSteps(initialSteps);
+    setSteps(initialSteps);
+  }, [testsData]);
+
+  const showModal = (value: string, group: number | string) => {
+    if (value === 'test') {
+      setIsTestModalVisible(true);
+    } else {
+      setIsDelayModalVisible(true);
     }
-  }, [initialSteps]);
 
-  const saveSteps = () => {
-    return updateTestSuite({
-      id: entityDetails.name,
-      data: {
-        ...entityDetails,
-        steps: currentSteps.map(step => {
-          return {
-            stopTestOnFailure: step.stopTestOnFailure,
-            execute: isV2
-              ? step.test
-                ? {name: step.test}
-                : {delay: step.delay}
-              : [step.test ? {test: step.test} : {delay: step.delay}],
-          };
-        }),
-      },
-    })
-      .then(res => displayDefaultNotificationFlow(res))
-      .then(() => {
-        notificationCall('passed', 'Steps were successfully updated.');
-      });
+    setCurrentGroup(group);
   };
 
-  const onSelectStep = (value: string) => {
-    if (value === 'delay') {
-      setIsDelayModalVisible(true);
+  const removeNode = (id: string, group: number) => {
+    if (steps[group].length === 1) {
+      setSteps([...steps.slice(0, group), ...steps.slice(group + 1)]);
     } else {
-      const {name, type} = JSON.parse(value);
-
-      setCurrentSteps([
-        ...currentSteps,
-        {
-          test: name,
-          id: nanoid(),
-          type,
-          stopTestOnFailure: false,
-        },
-      ]);
+      setSteps([...steps.slice(0, group), steps[group].filter(item => item.id !== id), ...steps.slice(group + 1)]);
     }
   };
 
-  const addDelay = (value: number) => {
-    setCurrentSteps([
-      ...currentSteps,
-      {
-        delay: `${value}ms`,
+  // add nodes
+  useEffect(() => {
+    const newNodes: ExtendedNode[] = [];
+
+    steps.forEach((step, index) => {
+      step.forEach((item: LocalStep, itemIndex: number) =>
+        newNodes.push({
+          type: 'test',
+          id: item.id!,
+          data: {...item, removeNode, group: index},
+          position: {x: index * 450, y: itemIndex * 100},
+          group: index,
+        })
+      );
+
+      newNodes.push({
+        type: 'intersection',
         id: nanoid(),
-        stopTestOnFailure: false,
-      },
-    ]);
-    setIsDelayModalVisible(false);
-  };
+        position: {x: index * 450 + 135, y: step.length * 100 - 20},
+        data: {showModal, group: index, withoutHandles: true},
+      });
 
-  const deleteStep = (index: number) => {
-    setCurrentSteps([...currentSteps.slice(0, index), ...currentSteps.slice(index + 1)]);
-  };
+      const intersectionGroup = `${index}-intersection`;
+      newNodes.push({
+        type: 'intersection',
+        id: nanoid(),
+        position: {x: (index + 1) * 450 - 90, y: 11},
```

Comment:
Probably building the position (or even node), should be extracted out

Reply:
Done